### PR TITLE
POIs

### PR DESCRIPTION
**Pull request by marc7s**
* Added POI support
* Support for adding POIs through scripts
* Added Parking POIs with parking and unparking support
* Added BusStop POI with support for stopping at bus stops
* Added a new NavigationMode to navigate between POIs
* Added multiple target support to navigation
* Added directional navigation graph support
* Added overrides for TargetLines and NavigationMode to CarSpawner
* Added support for buses, with manual creation of the routes
* Added different VehicleTypes for the vehicles
* The new navigation mode navigates to different POIs depending on the VehicleType
* Fixed the NavigationPath line in intersections so it is still straight

### CODE REVIEW

**Comment by hannes44** [comment]
LGTM!!!!!!!

**Comment by hannes44** [comment]
Censur 😠

**Comment by hannes44** [comment]
🤓

**Comment by hannes44** [comment]
👁️ 👅 👁️

File: TrafficSimulator/Assets/RoadGenerator/Core/Runtime/DataModel/POIs/Parking.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        public GameObject ParkNodePrefab;
+
+        [Header("Settings")]
+        public ParkingLineup ParkingLineup = ParkingLineup.Random;
+
+        [Header("Debug Settings")]
+        public bool DrawParkNodes = false;
+        
+        private List<POINode> _parkingSpots = new List<POINode>();
+        private const float parkingSize = 5f;
```
Comment (hannes44) [inline TrafficSimulator/Assets/RoadGenerator/Core/Runtime/DataModel/POIs/Parking.cs:26]: Coding standard

File: TrafficSimulator/Assets/RoadGenerator/Core/Runtime/DataModel/POIs/Parking.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            Size = new Vector3(20, 0.1f, 20);
+            GenerateParkingSpots();
+            ShowParkNodes();
+        }
+
+        private void GenerateParkingSpots()
+        {
+            _parkingSpots.Clear();
+            float sideOffset = Size.z / 2;
+            Vector3 startPos = transform.position + transform.right * sideOffset;
```
Comment (hannes44) [inline TrafficSimulator/Assets/RoadGenerator/Core/Runtime/DataModel/POIs/Parking.cs:49]: Add space

File: TrafficSimulator/Assets/RoadGenerator/Core/Runtime/DataModel/Tram.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        // TODO: Add TramStops
+        // List<TramStop> tramRoute;
+        // Start is called before the first frame update
+        void Start()
+        {
+            base.Init();
+        }
+
+        // Update is called once per frame
```
Comment (hannes44) [inline TrafficSimulator/Assets/RoadGenerator/Core/Runtime/DataModel/Tram.cs:17]: Remove update

File: TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/POINode.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        
+        /// <summary>Tries to assign a vehicle to this node. Returns `true` if it succeded, `false` if there is already a vehicle assigned</summary>
+        public virtual bool SetVehicle(Vehicle vehicle)
+        {
+            if(_vehicle == null || _vehicle == vehicle)
+            {
+                _vehicle = vehicle;
+                return true;
+            }
+            return false;
```
Comment (belgien) [inline TrafficSimulator/Assets/RoadGenerator/Core/Runtime/Objects/POINode.cs:36]:
```suggestion
            }
            
            return false;
```

**Review by belgien — APPROVED** [review]

**Pull request merged**